src/cmake/usingOpenCV.cmake

```diff
@@ -1,26 +1,41 @@
-find_package(opencv CONFIG REQUIRED)
+find_package(opencv REQUIRED)
+if (opencv_FOUND)
+  set_property(TARGET opencv::opencv PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${DIRECTORIES})
+  target_link_libraries(${TARGET_NAME} PRIVATE opencv::opencv)
 
-# The following code is added to work around conan issues with Xcode not seeing the release version in the IDE and
-# giving false errors
-get_property(
-  INCLUDE_DIRECTORY
-  TARGET opencv::opencv
-  PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
-if(DEFINED opencv_INCLUDE_DIRS_DEBUG)
-  set(DIRECTORIES
-      "$<$<CONFIG:Debug>:${opencv_INCLUDE_DIRS_DEBUG}>" "$<$<CONFIG:Release>:${opencv_INCLUDE_DIRS_DEBUG}>"
-      "$<$<CONFIG:RelWithDebInfo>:${opencv_INCLUDE_DIRS_DEBUG}>" "$<$<CONFIG:MinSizeRel>:${opencv_INCLUDE_DIRS_DEBUG}>")
-else()
-  set(DIRECTORIES
-      "$<$<CONFIG:Debug>:${opencv_INCLUDE_DIRS_RELEASE}>" "$<$<CONFIG:Release>:${opencv_INCLUDE_DIRS_RELEASE}>"
-      "$<$<CONFIG:RelWithDebInfo>:${opencv_INCLUDE_DIRS_RELEASE}>"
-      "$<$<CONFIG:MinSizeRel>:${opencv_INCLUDE_DIRS_RELEASE}>")
-endif()
-set_property(TARGET opencv::opencv PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${DIRECTORIES})
+  # The following code is added to work around conan issues with Xcode not seeing the release version in the IDE and
+  # giving false errors
+  get_property(
+    INCLUDE_DIRECTORY
+    TARGET opencv::opencv
+    PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+  if(DEFINED opencv_INCLUDE_DIRS_DEBUG)
+    set(DIRECTORIES
+            "$<$<CONFIG:Debug>:${opencv_INCLUDE_DIRS_DEBUG}>"
+            "$<$<CONFIG:Release>:${opencv_INCLUDE_DIRS_DEBUG}>"
+            "$<$<CONFIG:RelWithDebInfo>:${opencv_INCLUDE_DIRS_DEBUG}>"
+            "$<$<CONFIG:MinSizeRel>:${opencv_INCLUDE_DIRS_DEBUG}>")
+  else()
+    set(DIRECTORIES
+            "$<$<CONFIG:Debug>:${opencv_INCLUDE_DIRS_RELEASE}>"
+            "$<$<CONFIG:Release>:${opencv_INCLUDE_DIRS_RELEASE}>"
+            "$<$<CONFIG:RelWithDebInfo>:${opencv_INCLUDE_DIRS_RELEASE}>"
+            "$<$<CONFIG:MinSizeRel>:${opencv_INCLUDE_DIRS_RELEASE}>")
+  endif()
+else ()
+  message(FATAL_ERROR "opencv not found")
+endif ()
 
-list(APPEND CUSTOM_DEPENDENCIES opencv::opencv)
+#set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
```

Comment:
Similarly, should this comment not be deleted?

Reply:
True! Thanks, removed.